apps/website/src/components/courses/exercises/AutoSaveTextarea.tsx

```diff
@@ -0,0 +1,243 @@
+import React, {
+  useCallback, useEffect, useState, useRef,
+} from 'react';
+import clsx from 'clsx';
+import SaveStatusIndicator from './SaveStatusIndicator';
+
+type SaveStatus = 'idle' | 'typing' | 'saving' | 'saved' | 'error';
+
+type AutoSaveTextareaProps = {
+  value: string;
+  onChange: (value: string) => void;
+  onSave: (value: string) => Promise<void>;
+  placeholder?: string;
+  minHeight?: string;
+  className?: string;
+  disabled?: boolean;
+  showSaveStatus?: boolean;
+  autoSaveDelay?: number; // milliseconds for inactivity timer (default 20000)
+  periodicSaveInterval?: number; // milliseconds for periodic saves (default 180000)
+  ariaLabel?: string;
+  hideResizeHandle?: boolean;
+  savedText?: string; // Custom text for saved state (default: "Answer saved")
+};
+
+const AutoSaveTextarea: React.FC<AutoSaveTextareaProps> = ({
+  value,
+  onChange,
+  onSave,
+  placeholder = 'Enter text here',
+  minHeight = '140px',
+  className,
+  disabled = false,
+  showSaveStatus = true,
+  autoSaveDelay = 20000, // 20 seconds
+  periodicSaveInterval = 180000, // 3 minutes
+  ariaLabel,
+  hideResizeHandle = false,
+  savedText = 'Answer saved',
+}) => {
+  const [saveStatus, setSaveStatus] = useState<SaveStatus>('idle');
+  const [lastSavedValue, setLastSavedValue] = useState<string>(value);
+  const isSavingRef = useRef<boolean>(false);
+  const inactivityTimerRef = useRef<number | null>(null);
+  const statusTimerRef = useRef<number | null>(null);
+
+  const isEditing = value !== lastSavedValue;
+
+  // Cleanup all timers on unmount
+  useEffect(() => {
+    return () => {
+      [inactivityTimerRef, statusTimerRef].forEach((ref) => {
+        if (ref.current) clearTimeout(ref.current);
+      });
+    };
+  }, []);
+
+  // Inject style for Firefox to hide drag notches if needed
+  useEffect(() => {
+    if (!hideResizeHandle) return undefined;
+
+    const styleId = 'auto-save-textarea-firefox-styles';
+    if (document.getElementById(styleId)) return undefined;
```

Comment:
This ID doesn't seem to be used anywhere, so I think this element will ever be found and we will never inject these custom styles (unless I'm missing where this is defined)